openfisca_core/holders.py

```diff
@@ -251,6 +270,71 @@ def graph(self, edges, get_input_variables_and_parameters, nodes, visited):
             return
         formula.graph_parameters(edges, get_input_variables_and_parameters, nodes, visited)
 
+    def get_memory_usage(self):
+        """
+            Gets data about the virtual memory usage of the holder.
+
+            :returns: Memory usage data
+            :rtype: dict
+
+            Exemple:
+
+            >>> holder.get_memory_usage()
+            >>> {
+            >>>    'nb_arrays': 12,  # The holder contains the variable values for 12 different periods
+            >>>    'nb_cells_by_array': 100, # There are 100 entities (e.g. persons) in our simulation
+            >>>    'cell_size': 8,  # Each value takes 8B of memory
+            >>>    'dtype': dtype('float64')  # Each value is a float 64
+            >>>    'total_nb_bytes': 10400  # The holder uses 10.4kB of virtual memory
+            >>>    }
+        """
+
+        usage = dict(
+            nb_cells_by_array = self.entity.count,
+            dtype = self.variable.dtype,
+            )
+
+        if self._array is not None:
+            # Only used when definition period is ETERNITY
+            usage.update(dict(
+                nb_arrays = 1,
+                total_nb_bytes = self._array.nbytes,
+                cell_size = self._array.itemsize,
+                ))
+            return usage
+        elif self._array_by_period is not None:
+            nb_arrays = sum([
```

Comment:
> In what case are there dictionnaries in `self._array_by_period` ?

Reply:
When the variable has "extra params". 

Extra params were introduced to be able to calculate the PPA, and the RSA from 2017. The way these benefits are calculated don't really match the OpenFisca model, as there are two periods involved: the "demand month" and the "running month". Hence the introduction of extra params. 

They for sure bring a lot of complexity for a single use case. @michelbl had ideas to remove them, but we never prioritized it.